proparse/src/main/java/org/prorefactor/proparse/antlr4/ProgressLexer.java

```diff
@@ -234,22 +302,34 @@ public String getArgText(String argName) {
   @Override
   public void undef(String argName) {
     // First look for a local file scoped defined name to undef
-    if (undef_helper(argName, (currentInclude.defdNames)))
+    if (currentInclude.isNameDefined(argName)) {
+      currentInclude.removeVariable(argName);
       return;
+    }
     // Second look for a named include arg to undef
     if (currentInclude.undefNamedArg(argName))
       return;
     // Third look for a non-local file scoped defined name to undef
     ListIterator<IncludeFile> it = includeVector.listIterator(includeVector.size());
     while (it.hasPrevious()) {
       IncludeFile incfile = it.previous();
-      if (undef_helper(argName, incfile.defdNames))
+      if (incfile.isNameDefined(argName)) {
+        incfile.removeVariable(argName);
         return;
+      }
     }
     // Last, look for a global arg to undef
     undef_helper(argName, globalDefdNames);
   }
 
+  private boolean undef_helper(String argName, Map<String, String> names) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS00100)